### PR TITLE
Basic tests

### PR DESCRIPTION
**Pull request by pierres**
Here are some basic tests which also can be run using Travis CI.

Also some bugs were found by these tests and fixed.

### CODE REVIEW

File: run [hunk 1]
```diff
@@ -15,16 +15,21 @@ options=''
 CACHE_CONTAINER=cache-container-$(pwd | sha1sum | awk '{print $1}')
 
 if ! docker ps -a --no-trunc | grep -q "${CACHE_CONTAINER}"; then
-    docker run -v ${HOME%/*} -v /root --name=${CACHE_CONTAINER} busybox chmod 1777 ${HOME%/*}
+    docker run -v ${HOME} -v /root --name=${CACHE_CONTAINER} busybox install -d -o $(id -u) -g $(id -g) ${HOME}
```
Comment (pierres) [inline run:18]: I cannot recall why I added the parent directory and not %HOME itself.

File: tests.bats [hunk 2]
```diff
@@ -0,0 +1,52 @@
+@test "--self-update repalces run executable" {
```
Comment (perprogramming) [inline tests.bats]: I like :)

File: run (hunk 1 above)
Comment (perprogramming) [reply run:18]: Was wondering here, too..

**Comment by perprogramming** [comment]
I love bats!

**Pull request merged**